common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -163,9 +163,16 @@ public void registerWithShuffleServer(
     client.sendRpc(removeBlocksMessage, new RpcResponseCallback() {
       @Override
       public void onSuccess(ByteBuffer response) {
-        BlockTransferMessage msgObj = BlockTransferMessage.Decoder.fromByteBuffer(response);
-        numRemovedBlocksFuture.complete(((BlocksRemoved)msgObj).numRemovedBlocks);
-        client.close();
+        try {
+          BlockTransferMessage msgObj = BlockTransferMessage.Decoder.fromByteBuffer(response);
+          numRemovedBlocksFuture.complete(((BlocksRemoved) msgObj).numRemovedBlocks);
+        } catch (Exception e) {
```

Comment:
Do you want to catch `Throwable` here?

Reply:
I haven’t seen any error could occur here, so I use Exception. Throwable of course is fine.